src/retention-management/nuxeo-retention-technical-overview.md

```diff
@@ -0,0 +1,166 @@
+
+
+- A S3 bucket dedicated to the records that is configured on [Compliance mode]({{page page='nuxeo-retention-technical-overview'}}#s3-configuration-requirements):,
+
+  - This bucket is used to store the records only,
+  - Compared to the standard bucket, the Garbage Collector must be disabled as there is a [specific deletion process]({{page page='nuxeo-retention-technical-overview'}}#record-deletion-flow),
+
+
+## {{> anchor 'dedicated-low-fields'}} Dedicated low-level fields
+
+In order to provide required data to Amazon S3 (expiration date and legal hold) and to ensure that records can’t be deleted when not eligible, some new low-level fields have been added at the storage level:
+
+- **ecm:isRecord** (boolean) - True if the document is declared as a record
+- **ecm:retainUntil** (datetime) - Expiration date (end of retention period)
+- **ecm:hasLegalHold** (boolean) - True if there is a legal hold applied to the record
+- **ecm:isUnderRetentionOrLegalHold** (boolean) - True if the record is under retention and/or legal hold
+
+If a legal hold is applied to a document or if a retention rule is applied to a document, then:
+
+- modification of the main blob (file:content) is prevented
+
+  - this includes modification through restoring an older version,
+
+
+- deletion (including recursive delete) is prevented:
+
+  - this is enforced at low-level,
+  - this information is made available per-document through a security policy,
+
+
+- ecm:isUnderRetentionOrLegalHold is set to true,
+
+This applies in all situations, no exception, including for Administrators.
+
+In case of [event-based retention]({{page page='nuxeo-retention-functional-overview'}}#create-retention-rule), the record is put on a temporary hold waiting for the event to occur (cf. [Retention flow]({{page page='nuxeo-retention-technical-overview'}}#retention-flow)). In this case, ecm:retainUntil is "indeterminate", so deletion is prevented as if it was a date far in the future (this is turned into a legal hold at the S3 storage level).
+
+A copy of a document resets its record, retention and legal hold information. This includes creating a new version (that’s why versioning must be disabled to be compliant).
+Theses fields don't apply to proxies, as proxies are just a convenience to access content stored elsewhere, and are not themselves records.
+
+## {{> anchor 'retention-flow'}} Retention flow
+
+When a granted user applies a retention rule to a document, the document is declared as a record, copy into the S3 compliance mode bucket.
+
+Then, depending on the retention rule type (immediate, metadata, or event-based retention), the expiration date is calculated and applied to both Nuxeo and S3 object.
+
+A background process checks when the event occurs and then starts the retention period (event-based retention).
+
+Another background process checks the document expiration to change the status and make the record deletable.
```

Comment:
Perhaps it is worth mentionning that this process is run once a day and there may be a delay the record is now deletable.